src/generate-simple-manifest.js

```diff
@@ -0,0 +1,51 @@
+const { validateGeneratedManifest } = require('./validate-generated-manifest.js')
+
+const generateSimpleManifest = function (manifest) {
+  // This function generates a simplified manifest with the container
+  // environment fields fully interpolated
+
+  const manifestErrors = validateGeneratedManifest(manifest)
+  if (manifestErrors.length) {
+    throw new Error(`Invalid manifest. errors=${JSON.stringify(manifestErrors, null, 2)}`)
+  }
+  const simpleManifest = {
+    manifest: {
+      name: manifest['manifest']['name'],
```

Comment:
This should use some kind of deep copy or clone. The current implementation will actually mutate the manifest that gets passed in, which would be pretty unexpected behavior for users of this module.

Reply:
The only object that needs to be deep copied is the containers field, right? The other fields are just strings.